Add instructions for building and running Fast-DDS on QNX 7.1 (#470)
eProsima/Fast-DDS-docs: docs/installation/sources/sources_qnx.rst

       cd $WORKSPACE/thirdparty/fastcdr
       git apply $WORKSPACE/build_qnx/qnx_patches/fastcdr_qnx.patch

       # Apply QNX patch to TinyXML2.
       # TinyXML2's CMakeLists.txt has CRLF, so use unix2dos to convert the patch to CRLF. 
       cd $WORKSPACE/thirdparty/tinyxml2
       unix2dos $WORKSPACE/build_qnx/qnx_patches/tinyxml2_qnx.patch
       git apply $WORKSPACE/build_qnx/qnx_patches/tinyxml2_qnx.patch

#. Get foonathan_memory_vendor:

   .. code-block:: bash

       cd $WORKSPACE
       vcs import < fastrtps.repos

#. Optional: clone GoogleTest and apply QNX patch to it:

    GoogleTest is required for building Fast-DDS tests.

   .. code-block:: bash

       cd $WORKSPACE
       git clone https://github.com/google/googletest.git && cd googletest
       git checkout v1.13.0
       git apply $WORKSPACE/build_qnx/qnx_patches/googletest_qnx.patch

#. Source the QNX environment script:

   .. code-block:: bash

       # We will assume the user's QNX SDP 7.1 is installed at ~/qnx710
       ~/qnx710/qnxsdp-env.sh

#. Build and install Fast-DDS and its dependencies:

    .. note::

       | To build examples, set COMPILE_EXAMPLES to ON in $WORKSPACE/build_qnx/common.mk.

>        | To build tests, set PERFORMANCE_TESTS, SYSTEM_TESTS, PROFILING_TESTS, and EPROSIMA_BUILD_TESTS to ON in $WORKSPACE/build_qnx/common.mk.

Reviewer: I think it would be best to refer to [Fast DDS' CMake option documentation](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html) to avoid listing too many variables.

    .. note::

       | All libraries will be installed to $(QNX_TARGET)/$(CPUVARDIR)/usr/lib.
       | All examples will be installed to $(QNX_TARGET)/$(CPUVARDIR)/usr/examples.
       | All tests will be installed to $(QNX_TARGET)/$(CPUVARDIR)/usr/bin/Fast-DDS_test.

       | QNX_TARGET is where your QNX SDP 7.1 installation's target folder is.
       | If QNX SDP 7.1 is installed at ~/qnx710, the QNX_TARGET will be at ~/qnx710/target/qnx7.
       | CPUVARDIR is a directory for a specific target architecture e.g. aarch64le and x86_64.

       | For example, libraries compiled for an aarch64 target will be at ~/qnx710/target/qnx7/aarch64le/usr/lib assuming QNX SDP 7.1 is installed at ~/qnx710.

   .. code-block:: bash

       cd $WORKSPACE/build_qnx
       make install -j 4

.. _run_examples_and_tests:

Run Examples and Tests on a QNX 7.1 Target
""""""""""""""""""""""""""""""""""""""""""

| Because examples and tests are compiled for QNX, they can only be run on a QNX target, not Ubuntu.

Move Libraries, Examples, and Tests to Your QNX Target
------------------------------------------------------

#. Move the built libraries to your QNX target:

>    We will assume that $(QNX_TARGET) is ~/qnx710/target/qnx7 and that $(CPUVARDIR) is aarch64le.

Reviewer: ```suggestion
   The following steps assume that $(QNX_TARGET) is ~/qnx710/target/qnx7 and that $(CPUVARDIR) is aarch64le. Adjust the values if this is not the case.
```

   .. code-block:: bash

       # Move Fast-CDR library to target
       scp ~/qnx710/target/qnx7/aarch64le/usr/lib/libfastcdr.so* root@<target-ip-address>:/usr/lib

       # Move Fast-DDS library to target
       scp ~/qnx710/target/qnx7/aarch64le/usr/lib/libfastrtps.so* root@<target-ip-address>:/usr/lib

       # Move Foonathan Memory library to target
       scp ~/qnx710/target/qnx7/aarch64le/usr/lib/libfoonathan_memory* root@<target-ip-address>:/usr/lib

       # Move TinyXML2 library to target
       scp ~/qnx710/target/qnx7/aarch64le/usr/lib/libtinyxml2.so* root@<target-ip-address>:/usr/lib

       # Move GoogleTest library to target
       scp ~/qnx710/target/qnx7/aarch64le/usr/lib/libgtest* root@<target-ip-address>:/usr/lib
       scp ~/qnx710/target/qnx7/aarch64le/usr/lib/libgmock* root@<target-ip-address>:/usr/lib

> #. Move examples and tests to your QNX target:

Reviewer: ```suggestion
#. Move examples and tests to the QNX target:
```

    .. code-block:: bash

       # Move Fast-CDR library to target
       scp -r ~/qnx710/target/qnx7/aarch64le/usr/examples root@<target-ip-address>:/var

       # Move Fast-DDS library to target
       scp -r ~/qnx710/target/qnx7/aarch64le/usr/bin/Fast-DDS_test root@<target-ip-address>:/var

Run DDSHelloWorldExample
------------------------

#. Open a terminal and run a subscriber:

    .. code-block:: bash

       # ssh into your target
       ssh root@<target-ip-address>

       # Run a subscriber
       /var/examples/cpp/dds/HelloWorldExample/bin/DDSHelloWorldExample subscriber
    
#. Open another terminal and run a publisher:

    .. code-block:: bash

       # ssh into your target
       ssh root@<target-ip-address>

>        # Run a subscriber

Reviewer: ```suggestion
       # Run a publisher
```

       /var/examples/cpp/dds/HelloWorldExample/bin/DDSHelloWorldExample publisher

> You should see the following output for subscriber:

Reviewer: ```suggestion
The following output will be shown in the subscriber terminal:
```

    .. code-block:: console

       Starting 
       Subscriber running. Please press enter to stop the Subscriber
       Subscriber matched.
       Message HelloWorld 1 RECEIVED
       Message HelloWorld 2 RECEIVED
       Message HelloWorld 3 RECEIVED
       Message HelloWorld 4 RECEIVED
       Message HelloWorld 5 RECEIVED
       Message HelloWorld 6 RECEIVED
       Message HelloWorld 7 RECEIVED
       Message HelloWorld 8 RECEIVED
       Message HelloWorld 9 RECEIVED
       Message HelloWorld 10 RECEIVED
       Subscriber unmatched.

> You should see the following output for publisher:

Reviewer: ```suggestion
The following output will be shown for the publisher:
```

    .. code-block:: console

       Starting 
       Publisher running 10 samples.
       Publisher matched.
       Message: HelloWorld with index: 1 SENT
       Message: HelloWorld with index: 2 SENT
       Message: HelloWorld with index: 3 SENT
       Message: HelloWorld with index: 4 SENT
       Message: HelloWorld with index: 5 SENT
       Message: HelloWorld with index: 6 SENT
       Message: HelloWorld with index: 7 SENT
       Message: HelloWorld with index: 8 SENT
       Message: HelloWorld with index: 9 SENT
       Message: HelloWorld with index: 10 SENT

Run a Test
----------

>     Because test binaries compiled for QNX cannot be run on Ubuntu, you must run the test binaries on a target which is running QNX.

Reviewer: ```suggestion
    Because test binaries compiled for QNX cannot be run on Ubuntu, test binaries must be run on a target which is running QNX.
```

    .. code-block:: bash

       # ssh into your target
       ssh root@<target-ip-address>

       # Run a test
       cd /var/Fast-DDS_test/unittest/dds/core/entity
       ./EntityTests

> You should see the following output for EntityTests:

Reviewer: ```suggestion
The following test output for EntityTests will be shown:
```
